### PR TITLE
feat: `StaticFileProviderFactory`

### PR DESCRIPTION
**Pull request by rkrasiuk**
## Motivation

Currently, with node builder it is impossible to be able to initialize state providers directly (e.g. `LatestStateProvider::new`) because `StaticFilesProvider` is not accessible.

## Description

Expose `StaticFilesProvider` by implementing `StaticFilesProviderFactory` for `BlockchainProvider` & `ProviderFactory` and add `StaticFilesProviderFactory` trait bound to `FullProvider`.

### CODE REVIEW

File: crates/stages/src/test_utils/macros.rs [hunk 1]
```diff
@@ -179,7 +179,7 @@ macro_rules! stage_test_suite_ext {
 
                 // Assert the successful result
                 let result = rx.await.unwrap();
-                runner.db().factory.static_file_provider().commit().unwrap();
+                reth_provider::StaticFileProviderFactory::static_file_provider(&runner.db().factory).commit().unwrap();
```
Comment (mattsse) [inline crates/stages/src/test_utils/macros.rs:182]: du we need `trait::fn(self)` here or can we just call `.factory.static_file_provider()`

**Review by mattsse — APPROVED** [review]
reasonable, one q re trait fn calls

File: crates/stages/src/test_utils/macros.rs (hunk 1 above)
Comment (rkrasiuk) [reply crates/stages/src/test_utils/macros.rs:182]: using full path since it's inside a macro

**Comment by DaniPopes** [comment]
image

**Pull request merged**